### DISCUSSION TITLE
problem with CMake and Boost

### DISCUSSION THREAD

**Discussion by shmayzer** [community/community#126973] [Programming Help]
### Body

tried to build this: https://github.com/jgaa/restc-cpp
by this instruction: https://github.com/jgaa/restc-cpp/blob/master/doc/GettingStarted.md
and there is an error on "cmake .." step: 

C:\Users\Artem\source\repos\nigalodeon\nigalodeon\restc-cpp\dbuild>cmake ..
-- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.22631.
-- Building restc-cpp version 0.99.0
-- EXTERNAL_RAPIDJSON_INCLUDE_DIR: C:/Users/Artem/source/repos/nigalodeon/nigalodeon/restc-cpp/dbuild/external-projects/src/externalRapidJson/include/rapidjson
-- Could NOT find GTest (missing: GTEST_LIBRARY GTEST_INCLUDE_DIR GTEST_MAIN_LIBRARY)
Will download and install googletest as a cmake included project
GTEST_TAG: main
-- Using internal log handler
-- Using C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.40.33807/bin/Hostx64/x64/cl.exe
-- Using C++ 17 for restc-cpp
CMake Error at C:/Program Files/CMake/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Boost (missing: system program_options filesystem date_time
  coroutine context chrono log) (found suitable version "1.85.0", minimum
  required is "1.58")
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files/CMake/share/cmake-3.29/Modules/FindBoost.cmake:2393 (find_package_handle_standard_args)
  CMakeLists.txt:217 (find_package)

-- Configuring incomplete, errors occurred!

What should i do?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Vinayak-YB** [comment]
This is because restc-cpp requires boost library as a dependency. You can try installing boost on your system using `sudo apt-get install libboost-all-dev ` or if its already installed then specify path during build `cmake .. -DBoost_INCLUDE_DIR= path to boost include -DBoost_LIBRARY_DIRS= path to boost lib`.
Ensure that the specified Boost version matches the one required by restc-cpp. As of latest it requires version 1.58.
